src/components/button/SplitButtonMenu.js

```diff
@@ -0,0 +1,112 @@
+
+  state = {
+    buttonLabel: this.firstChild.caption,
+    popoverActive: false,
+    popoverAnchorEl: null,
+  };
+
+  handleMainButtonClick = event => {
+    this.props.onButtonClick(event.currentTarget);
+  };
+
+  handleSecondaryButtonClick = event => {
+    this.setState({ popoverActive: true, popoverAnchorEl: event.currentTarget });
+  };
+
+  handleMenuItemClick = child => {
+    const childProps = child.props;
+    this.setState({
+      popoverActive: false,
+    });
+
+    if (this.props.triggerListAction) {
+      childProps.onMenuItemClick(childProps.caption);
+    } else {
+      this.setState({
+        buttonLabel: childProps.caption,
+      });
+    }
+  };
+
+  handleCloseClick = () => {
+    this.setState({ popoverActive: false });
+  };
+
+  render() {
+    const { children, level, size, triggerListAction, width, ...others } = this.props;
+    const { buttonLabel, popoverActive, popoverAnchorEl } = this.state;
+    const boxProps = pickBoxProps(others);
+    return (
+      <Box display="flex" justifyContent="center" {...boxProps} data-teamleader-ui="split-menu">
+        <ButtonGroup segmented>
+          <Button
+            label={buttonLabel}
+            level={level}
+            size={size}
+            onClick={this.handleMainButtonClick}
+            style={{ width: `${width}px` }}
+          />
+          <Button
+            icon={<IconChevronDownSmallOutline />}
+            level={level}
+            size={size}
+            onClick={this.handleSecondaryButtonClick}
+          />
+        </ButtonGroup>
+        <Popover
+          active={popoverActive}
+          anchorEl={popoverAnchorEl}
+          backdrop="transparent"
+          onEscKeyDown={this.handleCloseClick}
+          onOverlayClick={this.handleCloseClick}
+          position="start"
+        >
+          <Menu>
+            {React.Children.map(children, child => {
+              if (child.props.caption !== buttonLabel) {
+                return React.cloneElement(child, {
+                  onClick: () => this.handleMenuItemClick(child),
+                });
+              }
+            })}
+          </Menu>
+        </Popover>
+      </Box>
+    );
+  }
+}
+
+SplitButtonMenu.defaultProps = {
+  level: 'primary',
+  size: 'medium',
+  triggerListAction: true,
+};
+
+SplitButtonMenu.propTypes = {
+  /** The MenuItems we pass to our component. */
+  children: PropTypes.node.isRequired,
+  /** Determines which kind of button to be rendered. */
+  level: PropTypes.oneOf(['primary', 'secondary', 'destructive']),
+  /** Size of the button. */
+  size: PropTypes.oneOf(['small', 'medium', 'large']),
+  /** Boolean to decide if we want to fire directly after clicking on a list item. */
+  triggerListAction: PropTypes.bool,
+  /** The function executed, when we click on the main button. */
+  onButtonClick: PropTypes.func.isRequired,
+  /** Width of the main button in px. */
+  widht: PropTypes.string,
```

Comment:
typo

Reply:
Why do we even need this prop? Why not just an automatic width?

Reply:
Otherwise our button changes from width the moment our label gets changed.

Reply:
Maybe that also just proves that the whole changing button feature is a weird feature.

Anyway: now we have to provide a certain width, which is not really feasible since labels can have completely different widths in different languages. So you might show a button with a lot of "padding" by default...

I find it also very weird that we actually implemented this feature with a prop. So the same looking component can now have 2 different behaviors. Sometimes it will do the action instantly, other times it will change the buttons. Isn't that confusing for the user? 

Do we already have some examples where we would use this split button? Maybe that will help me to wrap my mind around this.

Reply:
The width prop was indeed a bad integration. To answer your second question i think the best way to explain this is by just linking you the guidelines(https://app.zeplin.io/project/599af68db2f4978a56dec839/screen/5cc1bc11bc9a132d5b674a50) for the split button.

Reply:
I still see this as a button with a default action and a lesser used secondary option

In the example of document layout templates:

- default action -> `download active template`
- click menu button -> secondary option `download example template`

Reply:
The component is updated so it always fires an action on a menu item click and the main action is always displayed in the primary button.